src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -0,0 +1,264 @@
+const ethers = require('ethers');
+const BigNumber = require('bignumber.js');
+const Web3 = require('web3');
+const {
+  Finding, FindingSeverity, FindingType, getJsonRpcUrl,
+} = require('forta-agent');
+
+// contract ABIs and network addresses
+const { getAbi, getAddress } = require('@uma/contracts-node');
+
+// UMA library functions
+const {
+  Networker,
+  Logger: logger,
+  createReferencePriceFeedForFinancialContract,
+} = require('@uma/financial-templates-lib');
+
+// load agent configuration
+const {
+  umaEverestId: UMA_EVEREST_ID,
+  optimisticOracle: optimisticOracleConfig,
+} = require('../../agent-config.json');
+
+const {
+  disputePriceErrorPercent,
+  cryptowatchApiKey: CRYPTOWATCH_API_KEY,
+  defipulseApiKey: DEFIPULSE_API_KEY,
+  tradermadeApiKey: TRADERMADE_API_KEY,
+  cmcApiKey: CMC_API_KEY,
+} = optimisticOracleConfig;
+
+const CHAIN_ID = 1; // mainnet
+
+// stores the optimistic oracle contract address
+let optimisticOracleAddress;
+
+logger.silent = true;
+
+// initialize global constants, web3 gets populated on initialization
+const getTime = () => Math.round(new Date().getTime() / 1000);
+const web3 = new Web3(new Web3.providers.HttpProvider(getJsonRpcUrl()));
+
+// create ethers interface object
+const optimisticOracleAbi = getAbi('OptimisticOracle');
+const iface = new ethers.utils.Interface(optimisticOracleAbi);
+
+/*
+Events we are interested in:
+
+    event RequestPrice(
+        address indexed requester,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes ancillaryData,
+        address currency,
+        uint256 reward,
+        uint256 finalFee
+    );
+
+    event ProposePrice(
+        address indexed requester,
+        address indexed proposer,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes ancillaryData,
+        int256 proposedPrice,
+        uint256 expirationTimestamp,
+        address currency
+    );
+*/
+const eventNames = ['RequestPrice', 'ProposePrice'];
+
+// calculate the % difference between 2 BigNumber values
+function calculatePercentError(first, second) {
+  const delta = first.minus(second).absoluteValue();
+  return delta.div(first).multipliedBy(100);
+}
+
+// creates a price feed using the UMA library
+async function createPriceFeed({ identifier, config }) {
+  // make a copy so we can change config w/o affecting the function argument (no-param-reassign)
+  const localConfig = config;
```

Comment:
I don't think this will operate the way the comment indicates.
```javascript
> config = {a: 5};
{ a: 5 }
> const localConfig = config;
undefined
> localConfig.b = 10
10
> config
{ a: 5, b: 10 }
```